dbhydra/dbhydra_core.py

```diff
@@ -760,11 +769,11 @@ class Joinable(Selectable):
 
 
 class AbstractTable(AbstractJoinable, abc.ABC):
-    def __init__(self, db1, name, columns = None, types = None, id_column_name = "id"):
+    def __init__(self, db1, name, columns = None, types: Optional[list[str]] = None, id_column_name: str = "id"):
         """id_column_name -> name of predefined column with autoincrement + PK"""
         super().__init__(db1, name, columns)
-        self.types = types
-        self.id_column_name = id_column_name
+        self.types: list[str] = [] if types is None else types
```

Comment:
you changed None to []

Reply:
When types are undefined I prefer to have None, not []

Reply:
does it have any reason that I dont see?

Reply:
Nope, I think that's a mistake. I got too hasty when type hinting.